sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -393,4 +407,13 @@ case class DataSourceResolution(
       nullable = true,
       builder.build())
   }
+
+  private def validateCatalog(catalogName: String): CatalogPlugin = {
+    try {
+      catalogManager.catalog(catalogName)
+    } catch {
+      case _: CatalogNotFoundException =>
```

Comment:
Seems it's OK to just throw `CatalogNotFoundException` here.

Reply:
Done